src/Data/Quantity.purs

```diff
@@ -91,30 +91,31 @@ instance eqQuantity ∷ Eq Quantity where
       u1 = derivedUnit q1'
       u2 = derivedUnit q2'
 
-instance showQuantity ∷ Show Quantity where
+instance Show Quantity where
   show (Quantity num unit) = show num <> " .* " <> show unit
 
 prettyDecimal ∷ Decimal → String
 prettyDecimal d =
-  if D.isInteger d && d < (D.fromNumber 1.0e18)
-    then D.toString d
-    else D.toString (D.toSignificantDigits 6 d)
+  D.toString
+    if D.isInteger d && d < D.fromNumber 1.0e18
+      then d
+      else D.toSignificantDigits 6 d
 
 -- | Show a physical quantity in a human-readable form, value and unit
 -- | separately.
 prettyPrint' ∷ Quantity → { number ∷ String, space ∷ Boolean, unit ∷ String }
 prettyPrint' (val .*. du)
   | du == unity = { number: prettyDecimal val, space: false, unit: "" }
-  | otherwise   = { number: prettyDecimal val, space: space, unit: toString du }
-    where space = du /= SI.degree
+  | otherwise   = { number: prettyDecimal val, space: du /= SI.degree, unit: toString du }
 
 -- | Show a physical quantity in a human-readable form.
 prettyPrint ∷ Quantity → String
 prettyPrint q =
-  case prettyPrint' q of
-    rec →
-      let space = if rec.space then " " else ""
-      in rec.number <> space <> rec.unit
+  if rec.space
+    then rec.number <> " " <> rec.unit
+    else rec.number <> rec.unit
```

Comment:
Did this to avoid an unnecessary string concatenation (premature optimization much?). This is also another alternative:
```suggestion
  rec.number <> if rec.space
                then " " <> rec.unit
                else rec.unit
```